src/main/java/de/unijena/cheminf/fragment/fingerprint/CountFingerprint.java

```diff
@@ -0,0 +1,179 @@
+    /**
+     * Returns the number of bits in this fingerprint.
+     * Since fragment fingerprints are key based, the number of bits in the fingerprint
+     * is equal to the number of predefined fragments (unique SMILES).
+     *
+     * @return the size of the fingerprint.
+     */
+    @Override
+    public long size() {
+        return this.fragmentArrayOfUniqueSmiles.length;
+    }
+    //
+    /**
+     * Returns the number of bins that are populated.
+     * The number is equal to the defined fragments.
+     *
+     * @return the number of populated bins
+     * @see    #size()
+     */
+    @Override
+    public int numOfPopulatedbins() {
+        return this.fragmentArrayOfUniqueSmiles.length;
+    }
+    //
+    /**
+     * For each defined structure in the fingerprint, returns the frequency.
+     * For an index greater than or equal to 0 and less than fragmentArrayOfUniqueSmiles, the frequency of this unique SMILES is output.
+     * If the given index is greater than the size of the fingerprint, a NullPinterException is thrown.
+     *
+     * @param index the index of the bin to return the number of hits for.
+     * @return the count for the bin with given index.
+     */
+    @Override
+    public int getCount(int index) {
+        if (index < this.fragmentArrayOfUniqueSmiles.length && index >= 0 && this.rawMap.containsKey(index)) {
+            return this.rawMap.get(index);
+        } else if (index >= this.fragmentArrayOfUniqueSmiles.length) {
+            return this.rawMap.get(index);
+        } else {
+            return 0;
+        }
+    }
+    //
+    /**
+     * Returns the hash corresponding to the given index in the fingerprint.
+     * Since this is a key-based fingerprint, the hash value is nothing
+     * more than the position of the bin in the fingerprint (no hash value is calculated).
+     *
+     * @param index the index of the bin to return the hash for.
+     * @return the hash for the bin with the given index.
+     */
+    @Override
+    public int getHash(int index) {
+        return this.uniqueSmilesToPositionMap.get(this.fragmentArrayOfUniqueSmiles[index]);
+    }
+    //
+    /**
+     *
+     * @param fp to be merged
+     */
+    @Override
+    public void merge(ICountFingerprint fp) {
+        throw new UnsupportedOperationException();
+    }
+    //
+    /**
+     *
+     * @param behaveAsBitFingerprint
+     */
+    @Override
+    public void setBehaveAsBitFingerprint(boolean behaveAsBitFingerprint) {
+        throw new UnsupportedOperationException();
+    }
+    //
+    /**
+     * Whether the fingerprint contains the given hash.
+     *The parameter hash is not a calculated hash value, but also corresponds to the position of the bin in the fingerprint.
+     * @see    #getHash(int)
+     * For hash greater than or equals fingerprint size the method returns false and for all hash less than fingerprint size a true is returned.
```

Comment:
"for all hash less than fingerprint size a true is returned"
The comment does not match the logic of the code